File: BookCatalogue/Front/project/src/app/models/book.model.ts
export class Book {
  public bookid: number;
  public name: string;
  public author: string;
  public year: number;

  constructor(bookid: number, name: string, author: string, year: number) {
    this.bookid = bookid;
    this.name = name;
    this.author = author;
    this.year = year;
  }
}

File: BookCatalogue/Back/BookCatalogue/src/main/java/com/example/apphandling/controller/model/BookModel.java
package com.example.apphandling.controller.model;
import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name = "book", schema = "book_database")
public class BookModel {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "book_id")
    private int bookid;
    @Column(name = "book_name")
    private String name;
    @Column(name = "book_author")
    private String author;
    @Column(name = "book_year")
    private String year;

}